apps/dashboard/src/components/editor/toolbar.js

```diff
@@ -24,22 +24,30 @@ import { ToolbarButton } from './styles/button';
 const Toolbar = ( { project } ) => {
 	const { saveEditorContent } = useDispatch( STORE_NAME );
 
-	const [ canRestoreDraft, editorContent, isSaving, isSaved ] = useSelect(
-		( select ) => [
-			filter( select( 'core/notices' ).getNotices(), {
-				id: UnpublishedChangesNotice.ID,
-			} ).length > 0,
-			select( 'core/block-editor' ).getBlocks(),
-			select( STORE_NAME ).isEditorSaving(),
-			select( STORE_NAME ).isEditorContentSaved(),
-		]
-	);
+	const [
+		canRestoreDraft,
+		editorContent,
+		isSaving,
+		isSaved,
+		projectTitle,
+	] = useSelect( ( select ) => [
+		filter( select( 'core/notices' ).getNotices(), {
+			id: UnpublishedChangesNotice.ID,
+		} ).length > 0,
+		select( STORE_NAME ).getEditorContent(),
+		select( STORE_NAME ).isEditorSaving(),
+		select( STORE_NAME ).isEditorContentSaved(),
+		select( STORE_NAME ).getEditorTitle(),
+	] );
 
 	const saveProject = () =>
-		saveEditorContent( project.id, parse( serialize( editorContent ) ) );
+		saveEditorContent( project?.id, parse( editorContent ), {
```

Comment:
So now that editor content is stored on `state.editor` we should be able to just call `saveEditorContent()` here. No need to fetch the data manually.

Same goes for `publishProject()`.

Reply:
If we want `saveEditorContent` to be callable without any params, all the content, title and extra options should come from somewhere. Content and title are now at `state.editor`, but `public` is something we do *on demand*.

I think this assumes that we already have the project ID stored on `state.editor`?

Reply:
Yeah. I think it makes sense to keep `public` as an option rather than in editor state as well - like, publishing in itself is more of an action than it is data.